src/wp-includes/wp-db.php

```diff
@@ -1327,7 +1327,7 @@ public function escape( $data ) {
 	 * @param string $string String to escape.
 	 */
 	public function escape_by_ref( &$string ) {
-		if ( ! is_float( $string ) && ! is_a( $string, WP_DB_Partial_Query::class ) ) {
+		if ( ! is_float( $string ) && ! $string instanceof WP_DB_Partial_Query ) {
```

Comment:
I think you can check the `$string instanceof WP_DB_Partial_Query` first. So whenever this method will trigger it will check if it's an instanceof WP_DB_Partial_Query & if not it will return. And if this condition is true we can check for the data type.